daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -232,9 +225,7 @@ final class ImmArray[+A] private (
   /** O(n) */
   def toArray[B >: A: ClassTag]: Array[B] = {
     val arr: Array[B] = new Array(length)
-    for (i <- indices) {
-      arr(i) = uncheckedGet(i)
-    }
+    Array.copy(array, start, arr, 0, length)
```

Comment:
What's the difference between `Array.copy` and `System.arraycopy`? And why use the former here but that latter in many other places?